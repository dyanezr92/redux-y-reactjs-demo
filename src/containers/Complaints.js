import { connect } from 'react-redux'
import { deleteComplaint,  addComplaint } from '../actions'
import UserTextSection from '../components/UserTextSection'
import { getComplaints } from '../reducers/complaintsReducer'

const mapStateToProps = (state) => ({
    elements: getComplaints(state),
    elementName: "complaints"
})

const mapDispatchToProps = (dispatch) => ({
  onDeleteClick:  (id) => dispatch(deleteComplaint(id)),
  onSubmit: ({text, author}) => dispatch(addComplaint({text, author}))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserTextSection)
