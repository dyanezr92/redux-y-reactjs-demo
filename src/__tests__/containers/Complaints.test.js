import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components
import { Provider } from 'react-redux';

import Complaints from '../../containers/Complaints';

// Configurar el mock de la store
const mockStore = configureStore();
const initialState = {
  comments: [],
  complaints: []
};
const store = mockStore(initialState);

// TEST
describe('<Complaints />', () => {
  test('shows the complaints with the store data', () => {
    const wrapper = shallow(
        <Complaints store={ store } />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
