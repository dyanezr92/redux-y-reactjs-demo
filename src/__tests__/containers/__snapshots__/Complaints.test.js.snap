// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Complaints /> shows the complaints with the store data 1`] = `
<UserTextSection
  elementName="complaints"
  elements={Array []}
  onDeleteClick={[Function]}
  onSubmit={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
